File: scripts/libs/ANIstatuses.js
module.exports = {
    poisoned: extend(StatusEffect, "dart-poisoned", {
        damage: 1.25,
        color: Pal.accent,
        speedMultiplier: 0.9,
        damageMultiplier: 0.4,
        healthMultiplier: 0.82,
        buildSpeedMultiplier: 0.06,
        dragMultiplier: 0.8,
        
        show: true,

        init(){
            this.super$init();
      
            this.opposite(StatusEffects.wet, StatusEffects.muddy);
        },
        load(){
            this.super$load();
            this.region = Core.atlas.find(this.name);
        },
        icons(){
            return this.region;
        },
        isHidden(){
            return false;
        },
        draw(unit){
            if(!unit.type.animalType){

                Draw.z(Layer.groundUnit + 1);
                Draw.mixcol(Pal.accent, Mathf.absin(2, 0.65));

                Draw.rect(unit.type.fullIcon, unit.x, unit.y, unit.rotation - 90);

                Draw.reset();

            } else unit.applyMixcol(Pal.accent, Mathf.absin(2, 0.65));
        }
    })
}

File: src/animalia/content/AniStatuses.java
package animalia.content;

import arc.graphics.g2d.Draw;
import arc.math.Mathf;
import mindustry.gen.Unit;
import mindustry.graphics.Layer;
import mindustry.graphics.Pal;
import mindustry.type.StatusEffect;

public class AniStatuses {
    public static StatusEffect poisoned;

    public static void load(){
        poisoned = new StatusEffect("poisoned"){{
                damage = 1.25f;
                color = Pal.accent;
                speedMultiplier = 0.85f;
                damageMultiplier = 0.65f;
                healthMultiplier = 0.85f;
                buildSpeedMultiplier = 0.8f;
                dragMultiplier = 0.8f;
            }

            @Override
            public void draw(Unit unit){
                Draw.z(Layer.groundUnit + 1);
                Draw.mixcol(Pal.accent, Mathf.absin(2, 0.65f));

                Draw.rect(unit.type.fullIcon, unit.x, unit.y, unit.rotation - 90);

                Draw.reset();
            }
        };
    }
}
